source/common/common/fine_grain_logger.cc

```diff
@@ -162,10 +161,22 @@ spdlog::logger* FineGrainLogContext::createLogger(const std::string& key)
   return new_logger.get();
 }
 
+void FineGrainLogContext::updateVerbosityDefaultLevel(level_enum level) {
+  absl::WriterMutexLock ul(&fine_grain_log_lock_);
```

Comment:
this is only a writer lock because we modify the verbosity_default_level, perhaps we can transition it to a reader lock when doing the loop to avoid blocking logging from occuring during this call